### PR TITLE
Implement RuntimeConfigFlags

### LINKED ISSUE #5880: ZIO 2.0: Deal with boolean blindness in RuntimeConfig
`RuntimeConfig` has a number of boolean fields. These can be tricky to get right and are not very stable in the presence of refactorings. Rather than continue this pattern of introducing more boolean flags, we can refactor to an alternate pattern: sets of case objects.

For example, instead of:

```scala
def foo(x: Boolean, y: Boolean, z: Boolean)
```

We can have:

```scala
final case class FooFlags(set: Set[FooFlag]) {
  def <> (that: FooFlags): FooFlags = ???
}
object FooFlags {
  val x: FooFlags = FooFlags(Set(FooFlag.X))
  val y: FooFlags = FooFlags(Set(FooFlag.X))
  val z: FooFlags = FooFlags(Set(FooFlag.X))
}
sealed trait FooFlag
object FooFlag {
  case object X extends FooFlag
  case object Y extends FooFlag
  case object Z extends FooFlag
}
```

Then the API becomes:

```scala
def foo(flags: FooFlags) = ???

foo(FooFlags.x <> FooFlags.y <> FooFlags.z)
```

In this case, we can have:

```scala
final case class RuntimeConfigFlags(...)
object RuntimeConfigFlags {
  val logRuntime: RuntimeConfigFlags = ...
}

sealed trait RuntimeConfigFlag
object RuntimeConfigFlag {
  case object ...
}
```

### PR DESCRIPTION
**Pull request by adamgfraser**
Resolves #5880.

### CODE REVIEW

File: core/shared/src/main/scala/zio/RuntimeConfigFlags.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package zio
+
+final case class RuntimeConfigFlags(flags: Set[RuntimeConfigFlag]) {
+
+  def +(flag: RuntimeConfigFlag): RuntimeConfigFlags =
+    RuntimeConfigFlags(flags + flag)
+
+  def apply(runtimeConfigFlag: RuntimeConfigFlag): Boolean =
```
Comment (jdegoes) [inline core/shared/src/main/scala/zio/RuntimeConfigFlags.scala:24]:
```suggestion
  def isEnabled(runtimeConfigFlag: RuntimeConfigFlag): Boolean =
```

**Review by jdegoes — COMMENTED** [review]
Slight preference for a rename, but otherwise, looks fantastic!

File: core/shared/src/main/scala/zio/RuntimeConfigFlags.scala (hunk 1 above)
Comment (adamgfraser) [reply core/shared/src/main/scala/zio/RuntimeConfigFlags.scala]: I like it!

**Comment by adamgfraser** [comment]
Done!

**Review by jdegoes — APPROVED** [review]

**Pull request merged**